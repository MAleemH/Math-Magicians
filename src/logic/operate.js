import Big from 'big.js';

const Operate = (numberOne, numberTwo, operation) => {
  const one = Big(numberOne);
  const two = Big(numberTwo);

  switch (operation) {
    case '+':
      return one.plus(two).toString();
    case '-':
      return one.minus(two).toString();
    case 'x':
      return one.times(two).toString();
    case 'รท':
      try {
        return one.div(two).toString();
      } catch (err) {
        return 'Enter Correct Values';
      }
    case '%':
      try {
        return one.mod(two).toString();
      } catch (err) {
        return 'Enter Correct Values';
      }
    default:
      throw Error(`Unknown Operation '${operation}'`);
  }
};

export default Operate;
