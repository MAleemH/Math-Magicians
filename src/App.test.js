import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Home from './components/home';
import Calculator from './components/calculator';
import Quote from './components/quote';
import Navbar from './components/navbar';
import NotMatch from './components/notMatch';
import operate from './logic/Operate.js';
import calculate from './logic/Calculate.js';

it('Home page renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Home />
      </BrowserRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Calculator page renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Calculator />
      </BrowserRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
it('Is Quotes renders correctly', () => {
  const tree = renderer
    .create(
      <Quote>
        <blockquote>
          <q className="quote-description">
            Software testing proves the existence of bugs not their absence.
          </q>
          <p className="quote-author">&mdash;Anonymous</p>
        </blockquote>
      </Quote>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Is NavBar renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    )
    .toJSON();
  expect(tree).toMatchInlineSnapshot(`
<nav
  className="navbar"
>
  <h1
    className="navTitle"
  >
    Math Magicians
  </h1>
  <ul
    className="navItems"
  >
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        className=""
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        className=""
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`);
});
it('Is NotMatch  renders correctly', () => {
  const tree = renderer
    .create(
      <NotMatch>
        <p>Test no to match render perfectly </p>
      </NotMatch>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Tests on operate.js arithmetic operations ', () => {
  test('Test multiply 2 * 5 to get 10', () => {
    const product = operate(2, 5, 'x');
    expect(product).toBe('10');
  });

  test('Test subtract 90 - 40 to get 50', () => {
    const subtraction = operate(90, 40, '-');
    expect(subtraction).toBe('50');
  });

  test('Test Add 100 + 30 to get 130', () => {
    const Addition = operate(100, 30, '+');
    expect(Addition).toBe('130');
  });

  test('Test divide 20 รท 2 to get 10', () => {
    const division = operate(20, 2, 'รท');
    expect(division).toEqual('10');
  });

  test('Test remainder 25 % 2 to get 3', () => {
    const modulo = operate(25, 2, '%');
    expect(modulo).toEqual('1');
  });
});

// CALCULATE TEST
describe('calculate', () => {
  test('AC button resets all values', () => {
    const initialState = {
      total: '5',
      next: '3',
      operation: '+',
    };
    const newState = calculate(initialState, 'AC');
    expect(newState).toEqual({
      total: null,
      next: null,
      operation: null,
    });
  });
  test('number button updates next value', () => {
    const initialState = {
      total: '5',
      next: '3',
      operation: '+',
    };
    const newState = calculate(initialState, '4');
    expect(newState).toEqual({
      total: '5',
      next: '34',
      operation: '+',
    });
  });
  test('operation button updates operation', () => {
    const initialState = {
      total: '5',
      next: '3',
      operation: '+',
    };
    const newState = calculate(initialState, 'x');
    expect(newState).toEqual({
      total: '8',
      next: null,
      operation: 'x',
    });
  });
  test('decimal point button updates next value', () => {
    const initialState = {
      total: '5',
      next: '3',
      operation: '+',
    };
    const newState = calculate(initialState, '.');
    expect(newState).toEqual({
      total: '5',
      next: '3.',
      operation: '+',
    });
  });
  test('equals button updates total value', () => {
    const initialState = {
      total: '5',
      next: '3',
      operation: '+',
    };
    const newState = calculate(initialState, '=');
    expect(newState).toEqual({
      total: '8',
      next: null,
      operation: null,
    });
  });
  test('negative button changes next value to negative', () => {
    const initialState = {
      total: '5',
      next: '3',
      operation: '+',
    };
    const newState = calculate(initialState, '+/-');
    expect(newState).toEqual({
      total: '5',
      next: '-3',
      operation: '+',
    });
  });
});
