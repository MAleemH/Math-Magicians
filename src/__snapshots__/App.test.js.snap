// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator page renders correctly 1`] = `
<div
  className="calculatorPage"
>
  <h2>
    Let's do some math!
  </h2>
  <div
    className="cal-container"
  >
    <p
      className="result"
    >
      0
    </p>
    <button
      className="keys"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="keys"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="keys"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="keys right-col"
      onClick={[Function]}
      type="button"
    >
      ÷
    </button>
    <button
      className="keys"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="keys"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="keys"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="keys right-col"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="keys"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="keys"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="keys"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="keys right-col"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="keys"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="keys"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="keys"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="keys right-col"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="bottom-zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="keys"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="keys right-col"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`Home page renders correctly 1`] = `
<div
  className="homepage"
>
  <h2>
    Welcome to our page!
  </h2>
  <p
    className="homepage-paragraphs"
  >
    <p>
      Lorem ipsum odor amet, consectetuer adipiscing elit. Sagittis consequat senectus adipiscing laoreet himenaeos. Lacus vulputate ex mauris vel velit ultricies mi. Leo risus laoreet dapibus class augue tortor? Parturient integer efficitur ac ultricies consectetur nibh. Ante feugiat tempus faucibus ante natoque blandit ornare. Phasellus imperdiet cras duis hendrerit massa nibh turpis consequat. Etiam sapien nunc orci ex viverra commodo vel. Dis primis commodo; class risus consectetur mauris eros elementum tristique. Leo aliquet molestie massa, pretium metus facilisi.
    </p>
    <p>
      Consectetur suspendisse mollis aptent massa neque dictum urna. Felis euismod a convallis praesent eleifend, parturient nascetur quis? Purus sit ipsum natoque ligula dolor vulputate blandit. Vulputate nibh magna habitant iaculis aptent dui est. Justo gravida lacus gravida pulvinar torquent ipsum. Sollicitudin bibendum facilisis enim; pretium taciti tellus viverra nibh. Facilisis feugiat quam conubia metus laoreet nostra iaculis auctor.
    </p>
  </p>
</div>
`;

exports[`Is NotMatch  renders correctly 1`] = `
<div>
  <p>
    Not Found!
  </p>
</div>
`;

exports[`Is Quotes renders correctly 1`] = `
<div
  className="quote-page"
>
  <blockquote>
    <q
      className="quote-description"
    >
      Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding.
    </q>
    <p
      className="quote-author"
    >
      —William Paul Thurston
    </p>
  </blockquote>
</div>
`;
